src/app/services/dataset.crud.service.ts

```diff
@@ -11,11 +11,18 @@
 */
 
 import { Injectable } from '@angular/core';
+import { Http } from '@angular/http';
+import { Observable } from 'rxjs/Observable';
 
 @Injectable()
 export class DatasetCrudService {
-  constructor(){}
+  constructor(private http: Http){}
 
+  private handleErrorObservable (error: Response | any) {
```

Comment:
The dataset CRUD service exists in "file.service.ts" lol I don't remember why it was named this way. But it has the dataset querying logic. You should keep one or the other, probably not best to keep both